### DISCUSSION TITLE
Conditional Forms: Stacked vs. nested fields recommendation

### DISCUSSION THREAD

**Discussion by waijeeho** [twilio-labs/paste#1493] [Office hours] [answered]
Hi team, was wondering if you guys had recommendations on how best to deal with forms -- specifically radio inputs -- that have content that is conditional.

Adding product & project background context in the Figma file here: https://www.figma.com/file/d2FZOlxT5F5FpxPqM9CfTr/Wires-Routing-on-Console-prototype?node-id=1543%3A48

**Question**
We are providing 3 options for inbound conversation routing. Developers can choose to:
1. Not route inbound conversations to any employees
2. Route inbound conversations to an employee based on custom logic within a callback URL
3. Route inbound conversations to a pool of employees, any of which will be assigned to the conversation

I chose the radio component to allow devs to choose one of the three options above, but options 2 & 3 have more nuances:
- If option 2 "custom routing" is selected, devs must add a callback URL for their custom logic.
- If option 3 "pool routing" is selected, we want developers to see the available pools.

I am not sure if for conditional forms, the best route is
1. Stack form fields and change the form underneath of the parent radio, like this:
![Screen Recording 2021-05-25 at 8 43 13 PM](https://user-images.githubusercontent.com/1030072/119586455-bdb06780-bd9a-11eb-836e-3e7ffc4f34f6.gif)
2. Or nest fields under the parent radio, similar to how it was solved in [this](https://github.com/twilio-labs/paste/discussions/1128) discussion:
![Screen Recording 2021-05-25 at 8 43 58 PM](https://user-images.githubusercontent.com/1030072/119586466-c30db200-bd9a-11eb-9ca1-e73c5a52ff69.gif)

I do not have user testing planned for this specific project, but I do want to test the console experience more broadly, as there are a few other areas I want to get feedback on.

Links: 
- Prototype:  https://www.figma.com/proto/d2FZOlxT5F5FpxPqM9CfTr/Wires-Routing-on-Console-prototype?page-id=968%3A10893&node-id=1009%3A10074&viewport=12211%2C-1125%2C0.25&scaling=min-zoom
   - Stacked form is under "1 3-option radio"
   - Nested form is under "UI Alternate 1"
- Design files:
   -  For stacked: https://www.figma.com/file/d2FZOlxT5F5FpxPqM9CfTr/Wires-Routing-on-Console-prototype?node-id=1295%3A396
   - For nested: https://www.figma.com/file/d2FZOlxT5F5FpxPqM9CfTr/Wires-Routing-on-Console-prototype?node-id=1468%3A239

Sorry if long-winded, but please let me know if there is any context that I can provide.

**Comment by SiTaggart** [comment, marked as answer]
Hi

Thanks for coming to Office Hours this week. Here is a link to the recording https://www.loom.com/share/17d88430c6cb40fead29a5dc4348ef91, the passcode for which can be found in [Slack](https://twilio.slack.com/archives/CBEQHHDLP/p1622245473021100)
